pkg/compiler/compiler.go

```diff
@@ -322,11 +322,15 @@ func (cs *LuetCompiler) buildPackageImage(image, buildertaggedImage, packageImag
 	// so the hash is unique also in cases where
 	// some package deps does have completely different
 	// depgraphs
-	// TODO: As the salt contains the packageImage ( in registry/organization/imagename:tag format)
-	// the images hashes are broken with registry mirrors.
-	// We should use the image tag, or pass by the package assertion hash which is unique
+	// TODO: We should use the image tag, or pass by the package assertion hash which is unique
 	// and identifies the deptree of the package.
 
+	imgRef := strings.SplitN(packageImage, "/", 2)
+	if len(imgRef) == 2 && helpers.IsDomainName(imgRef[0]) {
+		Debug(pkgTag, "Trim domain name from image reference", imgRef[0], "from", packageImage)
+		packageImage = imgRef[1]
```

Comment:
we shouldn't mutate `packageImage` at this stage, but rather have a new variable as an argument to L334 below. 

This is because `packageImage` is the image which we might have to push (if not present already), and we are computing here just a fingerprint to identify it later https://github.com/mudler/luet/blob/cc354552d950295e9d3af5f74f81bc43ab0de11c/pkg/compiler/compiler.go#L394.

Reply:
right makes sense, done.